src/julia_gc.c

```diff
@@ -0,0 +1,849 @@
+/****************************************************************************
+**
+*W  julia_gc.c
+**
+**  This file stores code only required by the Julia garbage collector
+**
+**  The definitions of methods in this file can be found in gasman.h,
+**  where the non-Julia versions of these methods live. See also boehm_gc.c
+**  and gasman.c for two other garbage collector implementations.
+**/
+
+#define JULIA_ENABLE_THREADING
+
+#include "code.h"
+#include "funcs.h"
+#include "gapstate.h"
+#include "gasman.h"
+#include "objects.h"
+#include "plist.h"
+#include "sysmem.h"
+#include "system.h"
+#include "vars.h"
+#include "fibhash.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "julia.h"
+#include "julia_gcext.h"
+
+#define MARK_CACHE_BITS 16
+#define MARK_CACHE_SIZE (1 << MARK_CACHE_BITS)
+
+#define MARK_HASH(x) (FibHash((x), MARK_CACHE_BITS))
+
+// #define STAT_MARK_CACHE
+
```

Comment:
Will this become a compile-time option?

Reply:
Good question. It's also not clear to me how useful that cache really is. Which reminds me: I asked @rbehrends before to add more comments to this file, I'll prod him again about it once this is merged, and/or add some more comments myself.

Reply:
Actually, I also just discovered that `MarkCache` doesn't actually seem to do anything! I.e. we never use it to prevent any computation. @rbehrends did we loose some code there, or what?